### DISCUSSION TITLE
Connection Errored Out

### DISCUSSION THREAD

**Discussion by pauliusx9** [lllyasviel/Fooocus#74] [Q&A] [answered]
Hey, I get an error a few seconds after I press Generate and therefore unable to generate anything. Any advice?

**Comment by EzmaMax23** [comment]
it may be a backend issue, a problem with your internet connection, an issue with your browser, IP address blockage

**Reply by pauliusx9** [reply]
By switching to Google Chrome from MS edge, seem to have fixed the issue. Thanks

**Reply by Ramit72** [reply]
Brother for me it didn't get fixed, I'm using Windows 11. Is yours still fine? and do you know how to fix it any other way

**Comment by La-Valentina** [comment]
Hi! It gives an error when trying to generate,
![2023-08-16_10-51-14](https://github.com/lllyasviel/Fooocus/assets/138053931/29de6b83-255a-43d5-9f49-b17c21d7d00b)
How to fix?

**Reply by tr88-ula** [reply]
I have the exact problem, how to fix please?

**Reply by EzmaMax23** [reply]
what are you trying to generate?

**Reply by La-Valentina** [reply]
Just a picture, it didn't generate any

**Reply by ConceptF1** [reply]
I have same problem.

**Reply by ConceptF1** [reply]
I found the solution from lllyasviel, who made Fooocus. 

erase and reinstall - It works!
The error occuered because of bad network.

**Comment by EzmaMax23** [comment]
Welcome

**Comment by EzmaMax23** [comment]
You can get the solution from user paul,he figured it out

**Reply by ConceptF1** [reply]
can you tell me where to find what paul said?

**Comment by biogwen** [comment]
I have also the problem and tried all the solutions found. I work fully local on my desktop, switched between the different browsers, I still have the error on the web page and a 1006 message on the backend terminal.

**Reply by alex57280** [reply]
I have exactly the same problem, I remove Avast, remove my VPN, nothing ...

**Comment by EzmaMax23** [comment]
Have you tried to restart it

**Comment by yuvrajg69** [comment]
![image](https://github.com/lllyasviel/Fooocus/assets/107845779/b1aad512-7ca9-48cd-b695-d6c3aeaeee87)
iam getting this error also this on my terminal
![Screenshot (313)](https://github.com/lllyasviel/Fooocus/assets/107845779/b190887e-6dd1-464e-80d7-93514e06a4ad)
please help

**Reply by grim-reapper** [reply]
Getting same error on window. Tried everything, nothing works..

**Reply by mcgillg3141** [reply]
Any luck? I'm having the same

**Reply by yuvrajg69** [reply]
Try google collab best solution

**Comment by lllyasviel** [comment, marked as answer]
several days ago we added the troubleshoots of common problems here:

https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md

`Connection error out` is included and hope that can be of help

**Reply by Ramit72** [reply]
My problem is still not fixed I'm on windows 11 and it is still doing the same error and now even before I Click on generate
![Fooocus 2 4 3 realistic - Google Chrome 10-07-2024 03_12_40 PM](https://github.com/lllyasviel/Fooocus/assets/175206349/13f12cd5-2634-4090-b1a3-6623f705bf10)

**Comment by ghost** [comment]
Hi. I'm experiencing this issue as well.

CMD:
C:\Users\ricar\Documents\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --preset realistic
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--preset', 'realistic']
Loaded preset: C:\Users\ricar\Documents\Fooocus\Fooocus\presets\realistic.json
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.844
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Total VRAM 6144 MB, total RAM 16310 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce GTX 1660 SUPER : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale'}
left over keys: dict_keys(['cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids'])
Base model loaded: C:\Users\ricar\Documents\Fooocus\Fooocus\models\checkpoints\realisticStockPhoto_v10.safetensors
Request to load LoRAs [['SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 0.25], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [C:\Users\ricar\Documents\Fooocus\Fooocus\models\checkpoints\realisticStockPhoto_v10.safetensors].
Loaded LoRA [C:\Users\ricar\Documents\Fooocus\Fooocus\models\loras\SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for UNet [C:\Users\ricar\Documents\Fooocus\Fooocus\models\checkpoints\realisticStockPhoto_v10.safetensors] with 788 keys at weight 0.25.
Loaded LoRA [C:\Users\ricar\Documents\Fooocus\Fooocus\models\loras\SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for CLIP [C:\Users\ricar\Documents\Fooocus\Fooocus\models\checkpoints\realisticStockPhoto_v10.safetensors] with 264 keys at weight 0.25.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cpu, use_fp16 = False.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 2.63 seconds
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 3.0
[Parameters] Seed = 6766464421860298807
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] Hot woman, cute, intricate, highly detailed, elegant, dynamic, vibrant color, cinematic, professional, thought driven, sharp focus, cool, inspirational, brilliant, colorful, coherent, polished, complex, amazing composition, fine detail, perfect light, determined, emotional, beautiful, pretty, attractive, illuminated, loving,, delicate, iconic, fabulous, epic, stunning
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] Hot woman, cute, intricate, highly detailed, elegant, quality, dynamic, dramatic light, sharp focus, beautiful, modern, stylish, illuminated, cinematic, background, professional, gorgeous, best, fair, pretty, perfect, logical, rational, thought, imposing, epic, artistic, innocent, rich, complex, romantic, scenic, color, pure, coherent, creative
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (1152, 896)
Preparation time: 11.09 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.02916753850877285, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model

C:\Users\ricar\Documents\Fooocus>pause
Press any key to continue . . .

**Reply by Manuverma79** [reply]
I am having the exact same issue, after it starts loading model it just keeps giving the "Connection error out".
Please let me know if you fixed it.

**Reply by Skukska** [reply]
Same system configuration and same problem

**Comment by EzmaMax23** [comment]
I was told to try Google collab

**Reply by ghost** [reply]
I never worked with Google Collab. Is there an instruction video or link to an instruction?

**Reply by mashb1t** [reply]
You can check out this video: https://www.youtube.com/watch?v=He8tW3Nl6Gs
Colab is quite easy to use, just navigate to https://colab.research.google.com/github/lllyasviel/Fooocus/blob/main/fooocus_colab.ipynb (or click toe Colab button in the readme of this repo to use the latest version of Fooocus there.
After clicking the "Play" button and waiting for a few minutes, you'll be able to use Fooocus for image generation. Please be advised that you may need more RAM for certain features such as img2img, so you might have to use the Colab Pro plan to do this.

**Reply by ghost** [reply]
I still get the error "Connection errored out"... How can I use Fooocus?

**Reply by mashb1t** [reply]
Please elaborate what you were doing in Colab before the error happened and check if RAM was maxed out (see previous comment)

**Reply by ghost** [reply]
1: I clicked the Google Colab link in your previous reply
2: I clicked the Play button
3: Got this window and I clicked Run anyway
![1](https://github.com/lllyasviel/Fooocus/assets/144709204/ec3eb98d-19a7-4260-8a10-7a095c8d2e4c)

4: It started installling requirements
![2](https://github.com/lllyasviel/Fooocus/assets/144709204/8b3fdb4c-6c24-4750-99d7-9e3b7b2fd29b)

5: It started installling requirements
![3](https://github.com/lllyasviel/Fooocus/assets/144709204/f63f0c2d-9f4c-4eab-9a36-b8f3f869169f)

6: It started installling requirements
![4](https://github.com/lllyasviel/Fooocus/assets/144709204/f65ece2e-029a-4c71-b2ef-a0e45c177d7b)

7: Finished installing requirements
![5](https://github.com/lllyasviel/Fooocus/assets/144709204/f12939d2-0112-4923-a475-b14dd5865c35)

8: I clicked the link
![6](https://github.com/lllyasviel/Fooocus/assets/144709204/14174fce-1fac-497b-9cc0-623fde804ea1)

**Reply by mashb1t** [reply]
The last step is missing i guess, render an image with performance Extreme Speed (at least as of your screenshots).
This exact setup works for me without errors, which is why i assume that the issue should be in your network or your Browser, not Colab or Fooocus.
Please make sure to test in a different Browser supporting Websockets. Also make sure you have no network / ad blockers enabled such as Pi-hole or Adblockers for gradio.live domains. To test this, you can open the link on your Smartphone connected only to Cellular Network (not WLAN).

EDIT: On cellular networks, also LTE routers, you may have to use a STUN/TURN server or a NAT gateway to make Websockets work.
See https://en.wikipedia.org/wiki/STUN + https://community.cisco.com/t5/collaboration-knowledge-base/demystifying-nat-traversal-with-stun-turn-and-ice/ta-p/4766853

**Comment by KennyPowers91** [comment]
Hi, I am getting this error when I try to use the faceswap, anyone knows the solution? Thanks in advance

**Reply by mashb1t** [reply]
see https://github.com/lllyasviel/Fooocus/issues/1594

**Reply by KennyPowers91** [reply]
Thank you, I was using the Google Colab as I dont have a VRAM hardware and the solution was to upgrade to the Pro Plan. Thanks again for your answer

**Reply by florix91** [reply]
i upgraded to pro plan as well but i am still getting the error. I get the error when i try to mix imput prompts with inpaint/outpain like i was in a video youtube.
